sdks/typescript/src/apache_beam/worker/operators.ts

```diff
@@ -43,21 +43,22 @@ import {
   createSideInputInfo,
 } from "./pardo_context";
 
-// Trying to get some of https://github.com/microsoft/TypeScript/issues/8240
-export const NonPromise = null;
+export const isPromise = (x: any): x is Promise<any> => {
+  return x instanceof Promise;
```

Comment:
In my initial tests this was expensive to do on every element call for every DoFn, but maybe I was over-optimizing. (There also didn't seem to be a standard way of checking for promises...maybe that's changed?)